overlord/snapshotstate/backend/backend.go

```diff
@@ -157,6 +159,50 @@ func Filename(snapshot *client.Snapshot) string {
 	return filepath.Join(dirs.SnapshotsDir, fmt.Sprintf("%d_%s_%s_%s.zip", snapshot.SetID, snapshot.Snap, snapshot.Version, snapshot.Revision))
 }
 
+// EstimateSnapshotSize calculates estimated size of the snapshot.
+func EstimateSnapshotSize(si *snap.Info, usernames []string) (uint64, error) {
+	var total uint64
+	calculateSize := func(path string, finfo os.FileInfo, err error) error {
+		if finfo.Mode().IsRegular() {
+			total += uint64(finfo.Size())
+		}
+		return err
+	}
+
+	visitDir := func(dir string) error {
+		exists, isDir, err := osutil.DirExists(dir)
+		if err != nil {
+			return err
+		}
+		if !(exists && isDir) {
+			return nil
+		}
+		return filepath.Walk(dir, calculateSize)
+	}
+
+	for _, dir := range []string{si.DataDir(), si.CommonDataDir()} {
+		if err := visitDir(dir); err != nil {
+			return 0, err
+		}
+	}
+
+	users, err := usersForUsernames(usernames)
```

Comment:
I don't think we're testing a path when usernames is non-nil. Does such use case even make sense? I guess we should make a worst-case estimate.

Reply:
You're right, added such tests for completeness. It's not really uses right now with automatic snapshots as they always pass nil, which expands to all users in usersForUsernames. Thanks